src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -1202,29 +1202,44 @@ codet java_bytecode_convert_methodt::convert_instructions(
       statement=std::string(id2string(statement), 0, statement.size()-2);
     }
 
+    typet catch_type;
+
+    // Find catch blocks that begin here. For now we assume if more than
+    // one catch targets the same bytecode then we must be indifferent to
+    // its type and just call it a Throwable.
     auto it=method.exception_table.begin();
     for(; it!=method.exception_table.end(); ++it)
     {
       if(cur_pc==it->handler_pc)
       {
-        // at the beginning of a handler, clear the stack and
-        // push the corresponding exceptional return variable
-        stack.clear();
-        auxiliary_symbolt new_symbol;
-        new_symbol.is_static_lifetime=true;
-        // generate the name of the exceptional return variable
-        const std::string &exceptional_var_name=
-          id2string(method_name)+
-          EXC_SUFFIX;
-        new_symbol.base_name=exceptional_var_name;
-        new_symbol.name=exceptional_var_name;
-        new_symbol.type=typet(ID_pointer, empty_typet());
-        new_symbol.mode=ID_java;
-        symbol_table.add(new_symbol);
-        stack.push_back(new_symbol.symbol_expr());
+        if(catch_type!=typet() || it->catch_type==symbol_typet())
+          catch_type=symbol_typet("java::java.lang.Throwable");
```

Comment:
No, as we can get multiple distinct types caught at the same location, I believe for Java 8 multi-catch blocks and empty catch blocks.